src/rust/catmem/pipe.rs

```diff
@@ -39,36 +59,64 @@ pub struct Pipe {
 
 impl Pipe {
     /// Creates a new pipe.
-    pub fn new(buffer: SharedRingBuffer<u16>) -> Self {
-        Self {
+    pub fn create(name: &str) -> Result<Self, Fail> {
+        Ok(Self {
             state: PipeState::Opened,
-            eof: false,
-            buffer: Rc::new(buffer),
-        }
+            pipe_type: PipeType::Push,
+            buffer: SharedRingBuffer::<u16>::create(name, RING_BUFFER_CAPACITY)?,
+        })
+    }
+
+    pub fn open(name: &str) -> Result<Self, Fail> {
+        Ok(Self {
+            state: PipeState::Opened,
+            pipe_type: PipeType::Pop,
+            buffer: SharedRingBuffer::open(name, RING_BUFFER_CAPACITY)?,
+        })
     }
 
-    /// Sets the end of file flag in the target pipe.
-    pub fn set_eof(&mut self) {
-        self.eof = true;
+    pub fn close(&mut self) -> Result<(), Fail> {
+        self.get_next_state()
     }
 
-    /// Gets the value of he end of file flag in the target pipe.
-    pub fn eof(&self) -> bool {
-        self.eof
+    pub fn try_close(&mut self) -> Result<(), Fail> {
+        match self.buffer.try_enqueue(EOF) {
+            Ok(()) => {
+                self.get_next_state()?;
+                Ok(())
+            },
+            Err(_) => Err(Fail::new(libc::EAGAIN, "Could not push")),
```

Comment:
Could you follow the same pattern that we have been using in the code, where we always log an error message when we fail?